docs/architecture/adr-112-proposer-based-timestamps.md

```diff
@@ -183,58 +190,61 @@ The proposer will set the re-proposed block's `Header.Timestamp` as the `Proposa
 
 Finally, the validation logic authenticates the timestamps in the `LastCommit.CommitSig`.
 The cryptographic signature in each `CommitSig` is created by signing a hash of fields in the block with the voting validator’s private key.
 One of the items in this `signedBytes` hash is the timestamp in the `CommitSig`.
 To authenticate the `CommitSig` timestamp, the validator authenticating votes builds a hash of fields that includes the `CommitSig` timestamp and checks this hash against the signature.
-This takes place in the [VerifyCommit function](https://github.com/cometbft/cometbft/blob/e8013281281985e3ada7819f42502b09623d24a0/types/validation.go#L25).
+This takes place in the [`VerifyCommit` function](https://github.com/cometbft/cometbft/blob/1f430f51f0e390cd7c789ba9b1e9b35846e34642/types/validation.go#L26).
 
+<!---
 #### Remove unused timestamp validation logic
 
-`BFTTime` validation is no longer applicable and will be removed.
+`BFT Time` validation is no longer applicable and will be removed.
 This means that validators will no longer check that the block timestamp is a weighted median of `LastCommit` timestamps.
 Specifically, we will remove the call to [MedianTime in the validateBlock function](https://github.com/cometbft/cometbft/blob/4db71da68e82d5cb732b235eeb2fd69d62114b45/state/validation.go#L117).
 The `MedianTime` function can be completely removed.
 
 Since `CommitSig`s will no longer contain a timestamp, the validator authenticating a commit will no longer include the `CommitSig` timestamp in the hash of fields it builds to check against the cryptographic signature.
+--->
 
-#### Timestamp validation when a block has not received a polka
+#### Timestamp validation when a block has not received a Polka
 
-The [POLRound](https://github.com/cometbft/cometbft/blob/68ca65f5d79905abd55ea999536b1a3685f9f19d/types/proposal.go#L29) in the `Proposal` message indicates which round the block received a polka.
+The [`POLRound`](https://github.com/cometbft/cometbft/blob/1f430f51f0e390cd7c789ba9b1e9b35846e34642/types/proposal.go#L29) in the `Proposal` message indicates which round the block received a Polka.
 A negative value in the `POLRound` field indicates that the block has not previously been proposed on the network.
-Therefore the validation logic will check for timely when `POLRound < 0`.
+Therefore the validation logic will check for timely when `POLRound == -1`.
 
-When a validator receives a `Proposal` message, the validator will check that the `Proposal.Timestamp` is at most `PRECISION` greater than the current Unix time known to the validator, and at maximum `PRECISION + MSGDELAY` less than the current Unix time known to the validator.
+When a validator receives a `Proposal` message, it records it `proposalReceiveTime` as the current Unix time known to the validator.
```

Comment:
General comment. You call "validator" when explaining that a proposal is received. But, if I'm not wrong, this will be run by all full nodes, won't it?

Reply:
I agree. I'm always confused because of the way our code is organized: we run the same code (validator vs full node), the only difference being that, at the moment of voting, we skip that if not a validator.